packages/node/src/parseForPolicy.js

```diff
@@ -19,88 +18,81 @@ const {
   parse,
   inspectImports,
   codeSampleFromAstNode,
+  inspectEsmImports,
 } = require('lavamoat-tofu')
 const { checkForResolutionOverride } = require('./resolutions')
 
+/**
+ * @callback ResolveHook
+ * @param {string} requestedName
+ * @param {string} referrer
+ * @returns {string|null}
+ */
+
 // file extension omitted can be omitted, eg https://npmfs.com/package/yargs/17.0.1/yargs
-const commonjsExtensions = ['', '.js', '.cjs']
-const resolutionOmittedExtensions = ['.js', '.json']
+const jsExtensions = new Set(
+  /** @type {const} */ (['.cjs', '.mjs', '.js', '.json'])
+)
+const cjsResolutionOmittedExtensions = /** @type {const} */ (['.js', '.json'])
```

Comment:
This list is a) incomplete (`.node`) and b) does not apply to ES modules.  Should probably investigate its use further and prevent it from being used an an ESM context, if feasible.

Reply:
I think this list only covers for `require('./filename')` where `filename.js` exists. It has no reason to be used for esm and it's possible it already is not.

Reply:
yeah, I'm not sure how this is used offhand; it just seems to be exported.